tests/integration/helpers.py

```diff
@@ -497,3 +502,230 @@ def get_unit_by_index(app_name: str, units: list, index: int):
+) -> None:
+    """Write content to the file in the provided unit.
+
+    Args:
+        ops_test: The ops test framework
+        unit: THe unit in which to write to file in
+        path: The path at which to write the content to
+        content: The content to write to the file
+        container_name: The container where to write the file
+    """
+    pod_name = unit.name.replace("/", "-")
+
+    with tempfile.NamedTemporaryFile(mode="w") as temp_file:
+        temp_file.write(content)
+        temp_file.flush()
+
+        subprocess.run(
+            [
+                "kubectl",
+                "cp",
+                "-n",
+                ops_test.model.info.name,
+                "-c",
+                container_name,
+                temp_file.name,
+                f"{pod_name}:{path}",
+            ],
+            check=True,
+        )
+
+    # return_code, _, _ = await ops_test.juju(
+    #     "ssh", unit.name, "chown", f"{MYSQL_SYSTEM_USER}:root", path
+    # )
+    # assert return_code == 0
+
+
+async def read_contents_from_file_in_unit(
+    ops_test: OpsTest, unit: Unit, path: str, container_name: str = CONTAINER_NAME
+) -> str:
+    """Read contents from file in the provided unit.
+
+    Args:
+        ops_test: The ops test framework
+        unit: The unit in which to read file from
+        path: The path from which to read content from
+        container_name: The container where the file exists
+
+    Returns:
+        the contents of the file
+    """
+    pod_name = unit.name.replace("/", "-")
+
+    with tempfile.NamedTemporaryFile(mode="r+") as temp_file:
+        subprocess.run(
+            [
+                "kubectl",
+                "cp",
+                "-n",
+                ops_test.model.info.name,
+                "-c",
+                container_name,
+                f"{pod_name}:{path}",
+                temp_file.name,
+            ],
+            check=True,
+        )
+
+        temp_file.seek(0)
+
+        contents = ""
+        for line in temp_file:
+            contents += line
+            contents += "\n"
+
+    return contents
+
+
+async def ls_la_in_unit(
+    ops_test: OpsTest, unit_name: str, directory: str, container_name: str = CONTAINER_NAME
+) -> list[str]:
+    """Returns the output of ls -la in unit.
+
+    Args:
+        ops_test: The ops test framework
+        unit_name: The name of unit in which to run ls -la
+        path: The path from which to run ls -la
+        container_name: The container where to run ls -la
+
+    Returns:
+        a list of files returned by ls -la
+    """
+    return_code, output, _ = await ops_test.juju(
+        "ssh", "--container", container_name, unit_name, "ls", "-la", directory
+    )
+    assert return_code == 0
+
+    ls_output = output.split("\n")[1:]
+
+    return [
+        line.strip("\r")
+        for line in ls_output
+        if len(line.strip()) > 0 and line.split()[-1] not in [".", ".."]
+    ]
```

Comment:
leaning towards keeping as is: the code currently filters out the `.` and `..` directories and also it's easier to parse the directories on separate lines (instead of `ls -l` which has them directories in the same line). also, this is a very minor nit, and doesnt affect the integrity of the test - i'd rather merge as is than refactor and cause test failures/debugs